README.md

```diff
@@ -1,735 +1,1009 @@
+    *   It should update the `s_userCollateral` mapping for `msg.sender` to reflect how much ETH they sent the contract.
+    *   It should emit a `CollateralAdded` event.
+    *   Don't forget to revert if `msg.value` is zero using `Engine__InvalidAmount()`.
 
-Now let's reduce the sender's collateral (in the mapping) and send it back to their address.
+    <details markdown='1'>
+    <summary>💡 Hint: Adding Collateral</summary>
 
-Emit `CollateralWithdrawn` with the sender's address, the amount they withdrew and the `currentPrice` from the DEX.
+    This is a simple function that:
+    - Receives ETH via `msg.value`
+    - Updates a mapping to track how much ETH each user has deposited
+    - Emits an event for tracking
+    
+    Remember to:
+    - Check for zero value
+    - Use the existing mapping
+    - Include the current ETH price (in MyUSD) in the event
+    
+    <details markdown='1'>
+    <summary>🎯 Solution</summary>
 
-Excellent! Re-deploy your contract with `yarn deploy` but first shut down and restart `yarn chain`. We want to do a fresh deploy of all the contracts so that they each have correct constructor parameters. Now try out your methods from the front end and see if you need to make any changes.
+    ```solidity
+    function addCollateral() public payable {
+        if (msg.value == 0) revert Engine__InvalidAmount();
+        
+        s_userCollateral[msg.sender] += msg.value;
+        emit CollateralAdded(msg.sender, msg.value, i_oracle.getETHMyUSDPrice());
+    }
+    ```
+
+    </details>
+    </details>
+
+2.  **Implement `calculateCollateralValue(address user)`**
+    *   This function should return the total USD value of the ETH collateral held by a `user`.
+    *   Use `i_oracle.getETHMyUSDPrice()` to get the current price of ETH in MyUSD (it returns price with 1e18 precision).
+    *   The collateral amount `s_userCollateral[user]` is in wei (1e18 wei = 1 ETH).
+    *   Calculation: `(collateralAmount * ethPrice) / PRECISION`.
+
+    <details markdown='1'>
+    <summary>💡 Hint: Calculating Collateral Value</summary>
+    This function converts ETH to USD value:
+    - Get the user's ETH amount from the mapping
+    - Get the current ETH price from the oracle
+    - Multiply them together and divide by PRECISION
+    
+    Think about:
+    - Why we need to divide by PRECISION
+    - What units the oracle price is in
+    - What units the collateral amount is in
+    
+    <details markdown='1'>
+    <summary>🎯 Solution</summary>
+    ```solidity
+    function calculateCollateralValue(address user) public view returns (uint256) {
+        uint256 collateralAmount = s_userCollateral[user];
+        return (collateralAmount * i_oracle.getETHMyUSDPrice()) / PRECISION;
+    }
+    ```
+    </details>
+    </details>
 
-Don't forget to give yourself some ETH from the faucet!
+🚀 Go ahead and re-deploy your contracts with `yarn deploy --reset` and test your front-end to see if you can add collateral. 
```

Comment:
will they need the `--reset` if they've made contract changes?

Reply:
Yes. They still need it because we need all the contracts to be re-deployed. If it only deploys one then the constructor params for other contracts will still be pointing to old contracts.

Reply:
ah got it